music21/analysis/harmonicFunction.py

```diff
@@ -0,0 +1,290 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+# Name:         harmonicFunction.py
+# Purpose:      Mapping between Roman numeral figures and harmonic function labels
+#
+# Authors:      Mark Gotham
+#
+# Copyright:    Copyright © 2022 Michael Scott Cuthbert and the music21 Project
+# License:      BSD, see license.txt
+# ------------------------------------------------------------------------------
+
+from typing import Union
+import unittest
+
+from music21 import common
+from music21 import key
+from music21 import roman
+from music21 import scale
+
+from music21 import environment
+_MOD = 'analysis.harmonicFunction'
+environLocal = environment.Environment(_MOD)
+
+
+class FUNKTION(common.enums.StrEnum):
+
+    TONIKA_DUR = 'T'
+    TONIKA_DUR_PARALLELKLANG_MOLL = 'Tp'
+    TONIKA_DUR_GEGENKLANG_MOLL = 'Tg'
+
+    TONIKA_MOLL = 't'
+    TONIKA_MOLL_PARALLELKLANG_DUR = 'tP'
+    TONIKA_MOLL_GEGENKLANG_DUR = 'tG'
+
+    SUBDOMINANT_DUR = 'S'
+    SUBDOMINANT_DUR_PARALLELKLANG_MOLL = 'Sp'
+    SUBDOMINANT_DUR_GEGENKLANG_MOLL = 'Sg'
+
+    SUBDOMINANT_MOLL = 's'
+    SUBDOMINANT_MOLL_PARALLELKLANG_DUR = 'sP'
+    SUBDOMINANT_MOLL_GEGENKLANG_DUR = 'sG'
+
+    DOMINANT_DUR = 'D'
+    DOMINANT_DUR_PARALLELKLANG_MOLL = 'Dp'
+    DOMINANT_DUR_GEGENKLANG_MOLL = 'Dg'
+
+    DOMINANT_MOLL = 'd'
+    DOMINANT_MOLL_PARALLELKLANG_DUR = 'dP'
+    DOMINANT_MOLL_GEGENKLANG_DUR = 'dG'
+
+
+_functionFigureTuplesKeyNeutral = (
+
+    ('T', 'I'),
+    ('t', 'i'),
```

Comment:
The value of the key should be a `HarmonicFunction.TONIC_MINOR` type entry.

Reply:
Why is this a tuple of tuples and not a Dict?

Reply:
Seemed a better representation as they map in both directions equally (both entries are used as both key and value). 

Anyway, no problem – happy to recast as a dict. Done.